bin/gem_downloader

```diff
@@ -20,7 +20,24 @@ Dir.chdir(gp.target_dir) {
   gp.target_files.each { |n, v, an|
     path = sprintf(file_format, index, n, v)
     loop {
-      `curl -o #{path} -L http://rubygems.org/downloads/#{n}-#{v}.gem`
+      # Fetch the pre-compiled gem native extensions instead of building them
+      # from source if the version has prefix `gem-fetch`.
+      if v.start_with?('gem-fetch-')
+        actual_version = v.dup
+        fail if actual_version.slice!('gem-fetch-').nil?
+        output = `gem fetch #{n} --version #{actual_version}`.strip.split(' ')
+        fail 'Unexpected output: #{output}' unless output[0] == 'Downloaded'
```

Comment:
Can we add a comment to say what we expect the string to look like? From this, it seems like we're expecting `Downloaded {package_name}`, is that true?

Reply:
That was my original expectation btw. E.g., this works under the assumption that `stdout` of

```
gem fetch grpc --version <version>
```

on linux, is

```
Downloaded grpc-<version>-x86_64-linux
```

Reply:
Added a comment for clarification.

Reply:
Also changed the `fail` check per feedback offline.

`actual_version.slice!('gem-fetch-')` seems to return `gem-fetch-` instead of the remainder.